common/testcontainers/testcontainers.go

```diff
@@ -0,0 +1,175 @@
+package testcontainers
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/scroll-tech/go-ethereum/ethclient"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/postgres"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+// TestcontainerApps testcontainers struct
+type TestcontainerApps struct {
+	postgresContainer *postgres.PostgresContainer
+	l1GethContainer   *testcontainers.DockerContainer
+	l2GethContainer   *testcontainers.DockerContainer
+
+	// common time stamp.
+	Timestamp int
+}
+
+// NewTestcontainerApps returns new instance of TestcontainerApps struct
+func NewTestcontainerApps() *TestcontainerApps {
+	timestamp := time.Now().Nanosecond()
+	return &TestcontainerApps{
+		Timestamp: timestamp,
+	}
+}
+
+// StartPostgresContainer starts a postgres container
+func (t *TestcontainerApps) StartPostgresContainer() error {
+	if t.postgresContainer != nil && t.postgresContainer.IsRunning() {
+		return nil
+	}
+	postgresContainer, err := postgres.RunContainer(context.Background(),
+		testcontainers.WithImage("postgres"),
+		postgres.WithDatabase("test_db"),
+		postgres.WithPassword("123456"),
+		testcontainers.WithWaitStrategy(
+			wait.ForLog("database system is ready to accept connections").WithOccurrence(2).WithStartupTimeout(5*time.Second)),
+	)
+	if err != nil {
+		log.Printf("failed to start postgres container: %s", err)
+		return err
+	}
+	t.postgresContainer = postgresContainer
+	return nil
+}
+
+// StartL1GethContainer starts a L1Geth container
+func (t *TestcontainerApps) StartL1GethContainer() error {
+	if t.l1GethContainer != nil && t.l1GethContainer.IsRunning() {
+		return nil
+	}
+	req := testcontainers.ContainerRequest{
+		Image:        "scroll_l1geth",
+		ExposedPorts: []string{"8546/tcp", "8545/tcp"},
+		WaitingFor:   wait.ForHTTP("/").WithPort("8545").WithStartupTimeout(100 * time.Second),
+		Cmd:          []string{"--log.debug", "ANY"},
+	}
+	genericContainerReq := testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	}
+	container, err := testcontainers.GenericContainer(context.Background(), genericContainerReq)
+	if err != nil {
+		log.Printf("failed to start scroll_l1geth container: %s", err)
+		return err
+	}
+	t.l1GethContainer, _ = container.(*testcontainers.DockerContainer)
+	return nil
+}
+
+// StartL2GethContainer starts a L2Geth container
+func (t *TestcontainerApps) StartL2GethContainer() error {
+	if t.l2GethContainer != nil && t.l2GethContainer.IsRunning() {
+		return nil
+	}
+	req := testcontainers.ContainerRequest{
+		Image:        "scroll_l2geth",
+		ExposedPorts: []string{"8546/tcp", "8545/tcp"},
```

Comment:
```suggestion
		ExposedPorts: []string{"8545/tcp"},
```

Reply:
I think ws port is not necessary, can try removing it.

Reply:
In the previous program, port 8545 was used, so it was retained; port 8546 was used for container exploration of testcontainer, so it was retained.

<img width="1437" alt="image" src="https://github.com/scroll-tech/scroll/assets/38887641/0df4359f-23e0-42ae-a628-24b829f03cac">

<img width="1357" alt="image" src="https://github.com/scroll-tech/scroll/assets/38887641/bc3088e2-9110-42b5-be10-df8f87dbfd1b">